server/server.js

```diff
@@ -150,6 +151,35 @@ app.get(routes.GET_ALL_TWEETS, async (req, res, next) => {
 	})(req, res, next);
 });
 
+app.post(routes.UPDATE_PWD, async (req, res, next) => {
+	passport.authenticate("jwt", { session: false }, (err, user, info) => {
+		const [ok, reason] = validPasswordFormat(req.body.newPassword);
+		if (!ok) {
+			console.log(reason);
+			res.status(StatusCodes.BAD_REQUEST).json({ error: reason });
+			return;
+		}
+
+
+		bcrypt.hash(req.body.newPassword, saltRounds, function(hashErr, hash) {
+			if (hashErr) {
+				res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ message: messages.PASSWORD_UPDATE_FAIL });
+				return;
+			}
+
+
```

Comment:
We need to hash and compare the old password against the stored hashed password

Once that matches, we need to store the hashed new password